File: symfony/src/Entity/SprintItem.php
<?php

namespace App\Entity;

use App\Repository\SprintItemRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: SprintItemRepository::class)]
#[ORM\Table(name: 'sprint_item')]
class SprintItem
{
    // El id es asignado explícitamente (1..5) para mantener la paridad de contrato
    // con el lado Spring Boot. Por eso no hay estrategia de generación automática.
    #[ORM\Id]
    #[ORM\Column]
    private int $id;

    #[ORM\Column(length: 120)]
    private string $title;

    #[ORM\Column(length: 60)]
    private string $category;

    #[ORM\Column(length: 30)]
    private string $status;

    #[ORM\Column]
    private int $weight;

    public function __construct(int $id, string $title, string $category, string $status, int $weight)
    {
        $this->id = $id;
        $this->title = $title;
        $this->category = $category;
        $this->status = $status;
        $this->weight = $weight;
    }

    public function getId(): int
    {
        return $this->id;
    }

    public function getTitle(): string
    {
        return $this->title;
    }

    public function getCategory(): string
    {
        return $this->category;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function getWeight(): int
    {
        return $this->weight;
    }
}

File: springboot/src/main/java/com/sincrodev/sidebyside/domain/SprintItem.java
package com.sincrodev.sidebyside.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

/**
 * Sprint 2: entidad JPA persistida en MySQL.
 * El id se asigna explícitamente (1..5) para mantener la paridad de contrato
 * con el lado Symfony, por eso no usa @GeneratedValue.
 */
@Entity
@Table(name = "sprint_item")
public class SprintItem {

    @Id
    private int id;
    private String title;
    private String category;
    private String status;
    private int weight;

    protected SprintItem() {
        // requerido por JPA
    }

    public SprintItem(int id, String title, String category, String status, int weight) {
        this.id = id;
        this.title = title;
        this.category = category;
        this.status = status;
        this.weight = weight;
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getCategory() {
        return category;
    }

    public String getStatus() {
        return status;
    }

    public int getWeight() {
        return weight;
    }
}
